pkg/build/nodeimage/const_stargz_snapshotter.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nodeimage
+
+/*
+The default Service Account manifest for Stargz Snapshotter.
+Stargz Snapshotter watches all registry secrets in the cluster for performing registry authentication.
+*/
+
+const defaultStargzSnapshotterSAManifest = `
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: stargz-snapshotter
+  namespace: kube-system
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: stargz-snapshotter
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: stargz-snapshotter
+subjects:
+- kind: ServiceAccount
+  name: stargz-snapshotter
+  namespace: kube-system
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: stargz-snapshotter
+rules:
+- apiGroups: [""]
+  resources: ["secrets"]
+  verbs: ["list", "watch"]
```

Comment:
this grants all nodes the capability to list and watch secrets.
can you explain the requirements here - why does the snapshotter needs this?

Reply:
for the kind usage or anyone comfortable with the origin of the stargz binary (i.e. a trusted source code build), this is fine.
RBAC is limited here and does not allow limiting the client to only get the secret type corev1.SecretTypeDockerConfigJson. one has to add external admission for that.

Reply:
For shipping to production clusters this seems a bit problematic though (adding another per node thing listing these) -- this seems like a flaw in containerd-cri + plugins.

Reply:
This approach also won't work with the standard approach to giving kubelet on-disk credentials, which our users leverage.

Reply:
i agree, the way this is managed does feel like a workaround.

Reply:
We started to track this issue on https://github.com/containerd/stargz-snapshotter/issues/270.
How about limiting the lazy pulling functionality only to public repositores at this moment?

Reply:
This is fixed since stargz snapshotter v0.6.1. 
Since v0.6.1, additional kubeconfig is no longer needed for registry authentication because snapshotter acquires creds through CRI (containerd/stargz-snapshotter#323).